src/CustomerioTracking.tsx

```diff
@@ -78,8 +96,8 @@ class CustomerIO {
    * @param identifier unique identifier for a profile
    * @param body (Optional) data to identify a profile
    */
-  static identify(identifier: string, body?: Object) {
-    CustomerioReactnative.identify(identifier, body);
+  static identify(identifier: string, body?: UserTraits) {
```

Comment:
By changing the function signature's data type to a segment SDK specific data type, that makes me think that we are creating a breaking change for exiting customers and making them migrate to Segment specific data types.

If this is true that we would require a migration in customer's apps, I don't currently agree with this approach and think we need to keep these existing functions backwards compatible.

Reply:
Would second what Levi said, i think we can still have previous and new methods because `analytics` can handle both of the cases

Reply:
I have tested these methods with current sample apps without making any change in the code. The typecasting always succeed from JSON to UserTraits and has not been an issue. I believe we need not add separate existing methods for backward compatibility as everything works well with this change too.

Reply:
@ami-aman did you check in both TS and JS? because TS might give the type compatibility failure

Reply:
Thanks for mentioning that you tested this code in the sample apps. That's a great test for us. 

I think that this code would not compile for a customer if they have this in their app: 

```ts
let data: Object = {"foo": "bar"}
CustomerIO.track("id", data)
```

If a customer is doing this: 
```ts
CustomerIO.track("id", {"foo": "bar"})
```

then I think that would compile with this change. 

I suggest we test the first code sample above in the sample apps and make sure the compiler works.

Reply:
Thanks for the great explaination, Levi! I am adding the code snippet you shared and testing if it passes or not.

@Shahroz16 I have tested on JS but as I am testing further with the code snippet shared by Levi, I will also test the methods in a TS file.

Reply:
@levibostian @Shahroz16 I tested the shared code snippet in a TS file (`Object` belongs to typescript) and the data was successfully sent to CDP and then CIO workspace. Data out for this record can be found [here](https://fly.customer.io/workspaces/122348/pipelines/destinations/364/data-out). 

Hence, I have tested the code on a TS file too and works smooth. Is there anything else that you want me to test ? If not, then this PR is awaiting approval.

Reply:
We might have a misunderstanding from this review thread. Shahroz and I are concerned about the TS code that existing customers have in their apps being backwards compatible at compile time with `tsc`. We are not concerned that the data types are compatible at runtime sending data to CIO. 

Could you show us the TS code that you used for testing? That could help resolve this quicker.

Reply:
Sure. 

There is a small edit in my last response. I tested these methods on Ami app and not our current sample apps. I have been repeatedly getting errors in sample apps for unknown reasons. I spent a day fixing it and then looking at the priority and urgency of the tickets I moved to ami app that was functional. 

I am again on fixing the sample apps and once done I will push the code to this branch.